packages/taquito/test/contract/contractView.spec.ts

```diff
@@ -137,4 +139,23 @@ describe('ContractView test', () => {
     }
     done();
   });
+
+  it('Should be able to execute tzip4 views', async (done) => {
+    mockRpcClient.runView.mockResolvedValue({
+      "data": {"int": "100"}
+    });
+
+    const result = await mockRpcClient.runView({
```

Comment:
It seems that the test is more testing the mock behavior than the tzip4 execution itself.

Some suggestions:
The test should call the `read` method, which will internally call `mockRpcClient.runView`.
The test should ensure that the returned value is formatted properly (`"data": {"int": "100"} `will be transform to `100`)

Note that if no `chainId` is passed in the `read` method, a mock will be needed for  `mockRpcClient.getChainId`.
It might be great to have a test case where a `chainId `is passed in the `read` method and another where it is no.
The first test will call `mockRpcClient.getChainId` and the second will not.

_The advantage of accepting the chainId as a param is to reduce the number of calls made to the RPC_

Reply:
that's a good point, will amend